packages/teams-js/src/public/dialog/adaptiveCard/bot.ts

```diff
@@ -0,0 +1,66 @@
+import { sendMessageToParent } from '../../../internal/communication';
+import {
+  dialogTelemetryVersionNumber,
+  getDialogInfoFromBotAdaptiveCardDialogInfo,
+} from '../../../internal/dialogHelpers';
+import { ensureInitialized } from '../../../internal/internalAPIs';
+import { ApiName, getApiVersionTag } from '../../../internal/telemetry';
+import { isHostAdaptiveCardSchemaVersionUnsupported } from '../../../internal/utils';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../../constants';
+import { BotAdaptiveCardDialogInfo, DialogInfo } from '../../interfaces';
+import { runtime } from '../../runtime';
+import { DialogSubmitHandler } from '../dialog';
+
+/**
+ * Module for interaction with adaptive card dialogs that need to communicate with the bot framework
+ *
+ * @beta
+ */
+/**
+ * Allows an app to open an adaptive card-based dialog module using bot.
+ *
+ * @param botAdaptiveCardDialogInfo - An object containing the parameters of the dialog module including completionBotId.
+ * @param submitHandler - Handler that triggers when the dialog has been submitted or closed.
+ *
+ * @beta
+ */
+export function open(botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo, submitHandler?: DialogSubmitHandler): void {
```

Comment:
Why did bot not end up with its own folder, given that it is a subcapability of adaptiveCard? I don't feel strongly about it, except that I want to make sure I understand why we're making this decision so we can give clear guidance to the rest of the engineering team about when and why to make new subfolders

Reply:
I didn't give `bot` its own folder since it doesn't have any sub-capabilites of its own, so the only thing that would be in the folder would be the `bot.ts` file itself. The same goes for `url.bot` and `url.parentCommunication`. If you feel like it would make more sense to have all of the bottom level sub-capabilites  be in their own folder, I'd be happy to add them, though it felt a bit extraneous to me, since following that schema, each capability/sub-capability would get its own folder, which feels unnecessary. Let me know your thoughts.

Reply:
I think that's fine, and seems like clear guidance we could give to people working on this repo. If a capability has subcapabilities, put it into a folder.